astropy/io/fits/file.py

```diff
@@ -369,11 +369,17 @@ def readarray(self, size=None, offset=0, dtype=np.uint8, shape=None):
                         # using the ACCESS_COPY mode in mmap so that users can
                         # modify arrays. However, on some systems, the OS raises
                         # a '[Errno 12] Cannot allocate memory' OSError if the
-                        # address space is smaller than the file. The solution
+                        # address space is smaller than the file. Also on windows
+                        # a '[WinError 1455] The paging file is too small for
+                        # this operation to complete' Windows error is raised or
+                        # equiivalent a '[Errno 22] Invalid argument. The solution
                         # is to open the file in mode='denywrite', which at
                         # least allows the file to be opened even if the
                         # resulting arrays will be truly read-only.
-                        if exc.errno == errno.ENOMEM and self.mode == "readonly":
+
+                        if (
+                            exc.errno == errno.ENOMEM or exc.winerror == 1455
```

Comment:
@saimn - I'm surprised that the tests even pass if `.winerr` doesn't exist! But I guess we don't go through this branch mostly. In principle, I guess it would be something like `or exc.errno == errno.EINVAL and getattr(exc, "winerror", 0) == 1455`

Obviously, I can adjust, but I now worry that we are adding not properly tested... What do you think?

Reply:
yeah `.winerr` is probably executed only with the test on windows currently. In the linux case I guess we test this branch only with `errno == ENOMEM` so `.winerr` is not reached.
`getattr(exc, "winerror", 0)` seems fine, not ideal but I don't really see a better alternative...